Classes/Cache/DatabaseCache.php

```diff
@@ -199,6 +199,15 @@ public function getUrlFromCacheBySpeakingUrl($rootPageId, $speakingUrl, $languag
 				'', 'expire'
 		);
 
+		// More than one urldata entry? Then prefer those with cHash
+		if (count($rows) > 1) {
+			if (preg_match('/cHash/', json_encode($rows))) {
+				$rows = array_filter($rows, function ($row) {
+					return strpos($row['original_url'], 'cHash') !== false ? true : false;
```

Comment:
It is better to make sure that there are no duplicates but it is hardly possible after 2.0.15. Now we can only do some more evaluation of `$rowCandidate` to see if cHash is there and use that.

Try to avoid tricks with `json_encode`, etc and make core more obvious.